src/main.rs

```diff
@@ -199,22 +212,29 @@ async fn hello_world() -> Html<&'static str> {
 
 #[derive(Clone)]
 pub struct AppConfig {
-    github_max_creation_time: DateTime<FixedOffset>,
-    eth_check_nonce_at_block: String,
-    eth_min_nonce:            i64,
-    eth_rpc_url:              String,
+    github_max_creation_time:    DateTime<FixedOffset>,
+    eth_check_nonce_at_block:    String,
+    eth_min_nonce:               i64,
+    eth_rpc_url:                 String,
+    transcript_file:             PathBuf,
+    transcript_in_progress_file: PathBuf,
 }
 
 impl Default for AppConfig {
     fn default() -> Self {
+        let transcript =
+            env::var("TRANSCRIPT_FILE").unwrap_or_else(|_| "./transcript.json".to_string());
+        let transcript_progress = format!("{}.new", transcript);
         Self {
-            github_max_creation_time: DateTime::parse_from_rfc3339(
+            github_max_creation_time:    DateTime::parse_from_rfc3339(
                 constants::GITHUB_ACCOUNT_CREATION_DEADLINE,
             )
             .unwrap(),
-            eth_check_nonce_at_block: constants::ETH_CHECK_NONCE_AT_BLOCK.to_string(),
-            eth_min_nonce:            constants::ETH_MIN_NONCE,
-            eth_rpc_url:              env::var("ETH_RPC_URL").expect("Missing ETH_RPC_URL"),
+            eth_check_nonce_at_block:    constants::ETH_CHECK_NONCE_AT_BLOCK.to_string(),
+            eth_min_nonce:               constants::ETH_MIN_NONCE,
+            eth_rpc_url:                 env::var("ETH_RPC_URL").expect("Missing ETH_RPC_URL"),
+            transcript_file:             PathBuf::from(transcript),
+            transcript_in_progress_file: PathBuf::from(transcript_progress),
         }
```

Comment:
These should be done using the clap CLI/ENV parser in `crate::main::Options`. All of these should be configurable this way and none of them are compile time constants.